gprofiler/metadata/application_identifiers.py

```diff
@@ -0,0 +1,248 @@
+import configparser
+import os.path
+import re
+from abc import ABCMeta, abstractmethod
+from typing import List, Optional
+
+from granulate_utils.linux.ns import resolve_host_path
+from psutil import NoSuchProcess, Process
+
+from gprofiler.log import get_logger_adapter
+
+_logger = get_logger_adapter(__name__)
+
+
+_PYTHON_BIN_RE = re.compile(r"^python([23](\.\d{1,2})?)?$")
+
+
+def _is_python_m_proc(process: Process) -> bool:
+    """
+    Checks whether the process ran as "python -m ..." pattern.
+    """
+    args = process.cmdline()
+
+    # Just to be on the safe side, module name must be run after the "-m" argument, python enforces it.
+    if len(args) < 3:
+        return False
+
+    return _is_python_bin(args[0]) and args[1] == "-m"
```

Comment:
Why aren't you using `_get_cli_arg_by_name(process.cmdline(), "-m")` here? (Which is "stronger" as it works on args in other indices)

Reply:
Well, rest of the code assumes that for `_is_python_m_proc` the returned process has the module at `argv[2]`... so meh.